app/scripts/lib/experiments/base.js

```diff
@@ -0,0 +1,164 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// An experiment interface.
+
+define([
+  'underscore',
+  'lib/storage',
+  'lib/url'
+],
+function (_, Storage, Url) {
+  'use strict';
+
+  var FORCE_GROUP_TYPE = 'forceExperimentGroup';
+  var storage = Storage.factory('localStorage');
+
+  function BaseExperiment() {
+    // nothing to do.
+  }
+
+  _.extend(BaseExperiment.prototype, {}, {
+    /**
+     * Name of the experiment. Used in metrics and view logic.
+     */
+    _name: null,
+    /**
+     * The A/B testing group type, see https://en.wikipedia.org/wiki/A/B_testing for details.
+     */
+    _groupType: null,
+    /**
+     * Initialize the experiment
+     *
+     * @returns {boolean}
+     *  Returns 'true' when the experiment successfully initialized. Otherwise 'false'.
+     */
+    initialize: function () {
+      // should be extended
+      return false;
+    },
+    /**
+     * Setup the experiment environment
+     *
+     * @param {String} name - name of the experiment
+     * @param {Object} options
+     * @returns {boolean}
+     *  Returns 'true' when the experiment successfully started. Otherwise 'false'.
+     */
+    start: function (name, options) {
+      // all experiments require these options
+      if (! name || ! options || ! options.window || ! options.metrics || ! options.able || ! options.user) {
+        return false;
```

Comment:
> Returns 'true' when the experiment successfully started. Otherwise 'false'.

It will just `false` and that signals that experiment failed to start up. which means NullExperiment will be used